### DISCUSSION TITLE
This Version of SAM vs Original SAM

### DISCUSSION THREAD

**Discussion by IFireflyl** [syntax-tm/SteamAchievementManager#21] [General]
I have been using the original version of SAM for years. I know it hasn't been updated in half a decade, but it also just works so I'm not sure what updates would be needed (other than maybe the GUI being modernized, and possibly having a working Search so you don't have to scroll through all your games to find the one you need).

Can you explain some of the benefits for using this version of SAM over the original? Additionally, if this version does have significant advantages or QoL improvements over the original, you may want to add something on the front page that explains this so people understand why they should choose this version. If there's something that you have done to improve this over the original, I will certainly be happy to give it a shot!

**Comment by syntax-tm** [comment]
Heyo,

Recently the focus has been on supporting people with 13,000+ games in their library. I have no idee if the old one can handle it or not, but this version **can** (_as of last week or so_) and there's still lots of improvements to be made.

I'm surprised whenever anyone uses this. It's just a thing I started a few years ago that I've kept up with and it's come a long way but there's still a lot I'd like to do. Any testing or feedback, good or bad, is very much appreciated. There's a lot of changes aside from a new UI framework though.

## Features

- Ability to add favorites
- Ability to hide certain apps
- Ability to export your library to JSON
- App shortcuts
    - Launch App
    - View on SteamDB
    - View on SteamGridDB
    - View on Steam Store
    - View on PCGamingWiki
    - View on Steam Card Exchange
- Real-time validation when editing stats

## General

- Completely new UI built using Windows Presentation Foundation
    - Support for hardware acceleration, vector graphics, better font scaling, etc.
- Upgraded to **.NET 8** from .NET Framework 4.0

## API

- Updated Steamworks interface implementations

## In-Progress

- Grid view for library
- Option to display only local images
- Option to display no images
- Ability to group apps
- Command line utility for automating and integrating with other applicaitons

## Planned

- Publish API project on NuGet
- Install/uninstall games and DLC
- Force reset all stats
- Theme support

## Transparency

One last thing is that unlike the old one, the new one is built here on GitHub. Everything, including the releases, are created using GitHub Workflows.

[Actions](https://github.com/syntax-tm/SteamAchievementManager/actions)

[Releases](https://github.com/syntax-tm/SteamAchievementManager/releases)

**Reply by IFireflyl** [reply]
Thank you for the detailed explanation! I don't have such a large library, but the other features you mentioned (both implemented and planned) intrigue me. I appreciate you taking the time to educate me. 😁
